internal/eval/fold.go

```diff
@@ -0,0 +1,242 @@
+package eval
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/cedar-policy/cedar-go/internal/ast"
+	"github.com/cedar-policy/cedar-go/types"
+)
+
+func foldPolicy(p *ast.Policy) *ast.Policy {
```

Comment:
It would be useful to have a comment here explaining the intuition behind what you're doing.

Reply:
Thanks - I've added some explanation, tell me if it seems sufficient.